### ISSUE TITLE
Does not work if d.dubBuildType is set

### ISSUE THREAD

**Issue by zorael** [Pure-D/serve-d#22]
Arch/Manjaro 64-bit, vscode is 1.19.2-1 from aur/code.

If I set `d.dubBuildType`, for instance to "`plain`", serve-d will stop working.

```d
2018-01-12T00:21:38.795:app.d:__lambda11:248 Processing as notification
2018-01-12T00:21:38.795:app.d:__lambda11:253 Failed processing notification: std.json.JSONException@std/json.d(260): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x38cd7f43]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x38cd7ebd]
??:? inout pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) std.json.JSONValue.objectNoRef() [0x38c7ab35]
??:? const pure @safe const(std.json.JSONValue)* std.json.JSONValue.opBinaryRight!("in").opBinaryRight(immutable(char)[]) [0x388c572f]
??:? bool workspaced.com.dub.setBuildType(std.json.JSONValue) [0x3897dae0]
??:? void served.extension.changedConfig(immutable(char)[][]) [0x388e8988]
??:? void app.processNotify(served.protocol.RequestMessage) [0x388777c7]
??:? void app.main(immutable(char)[][]).__lambda11() [0x38878e5c]
??:? void core.thread.Fiber.run() [0x38cbbb97]
??:? fiber_entryPoint [0x38cbba66]
??:? [0xffffffff]
```

Hovering over something only makes it say "DCD isn't initialized yet" in the output console.

```
2018-01-12T00:22:01.500:app.d:__lambda10:234 Responding with: ResponseMessage(7, null, const(ResponseError)(serverNotInitialized, "DCD isn't initialized yet", null))
{"code":-32002,"data":null,"message":"DCD isn't initialized yet"}
2018-01-12T00:22:01.500:jsonrpc.d:send:66 {"error":{"code":-32002,"data":null,"message":"DCD isn't initialized yet"},"id":7,"jsonrpc":"2.0"}
[Error - 00:22:01] Request textDocument/hover failed.
  Message: DCD isn't initialized yet
  Code: -32002 
```

Removing the `d.dubBuildType` setting makes it work again.

Attaching the full log: [`serve-d.log`](https://github.com/Pure-D/serve-d/files/1624484/serve-d.log)

**Comment by WebFreak001** [comment]
Actually the relevant line is
```
2018-01-12T16:16:01.265:app.d:__lambda11:249 Failed processing notification: std.json.JSONException@std/json.d(259): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x6d457af9]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x6d457a75]
??:? std.json.JSONValue.objectNoRefinout(pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) function()) [0x6d3fe245]
??:? std.json.JSONValue.opIndexinout(pure ref @safe inout(std.json.JSONValue) function(immutable(char)[])) [0x6d3fe410]
??:? void served.extension.changedConfig(immutable(char)[][]) [0x6d0816e3]
??:? void app.processNotify(served.protocol.RequestMessage) [0x6d0300a5]
??:? void app.main(immutable(char)[][]).__lambda11() [0x6d032560]
??:? void core.thread.Fiber.run() [0x6d439ccf]
??:? fiber_entryPoint [0x6d439bd2]
??:? [0xffffffff]
```
I will debug this later today

**Comment by WebFreak001** [comment]
fixed in dc31e0ca
